addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -622,6 +626,67 @@ options.registry.Theme = options.Class.extend({
 
         await this._reloadBundles();
     },
+    /**
+     * @see this.selectClass for parameters
+     */
+    async openCustomCodeDialog(previewMode, widgetValue, params) {
+        let websiteId;
+
+        let fieldName, contentText;
+        if (widgetValue === 'head') {
+            fieldName = 'custom_code_head';
+            contentText = _t('Enter code that will be added into the <head> of every page of your site.');
+        } else {
+            fieldName = 'custom_code_footer';
+            contentText = _t('Enter code that will be added before the </body> of every page of your site.');
+        }
+
+        this.trigger_up('context_get', {
+            callback: (ctx) => {
+                websiteId = ctx['website_id'];
+            },
+        });
+        const websites = await this._rpc({
+            model: 'website',
+            method: 'read',
+            args: [[websiteId], ['custom_code_head', 'custom_code_footer']],
+        });
+
+        await new Promise(resolve => {
+            const $content = $(core.qweb.render('website.custom_code_dialog_content', {
+                contentText,
+            }));
+            const aceEditor = this._renderAceEditor($content.find('.o_ace_editor_container')[0], websites[0][fieldName] || '');
+            const dialog = new Dialog(this, {
+                title: widgetValue,
```

Comment:
Instead of "head" and "footer", I would use the text you used as the field labels: "Custom <head> code" and "Custom end of <body> code". Declare those string the same way as `contentText`